applications/server/controllers/default.py

```diff
@@ -1,25 +1,56 @@
 # -*- coding: utf-8 -*-
-# this file is released under public domain and you can use without limitations
 
-#########################################################################
-## This is a sample controller
-## - index is the default action of any application
-## - user is required for authentication and authorization
-## - download is for downloading files uploaded in the db (does streaming)
-#########################################################################
+from gluon import utils as gluon_utils
+import time
+
 
-@auth.requires_login()
 def index():
     """
-    example action using the internationalization operator T and flash
-    rendered by views/default/index.html or views/generic.html
-
-    if you need a simple wiki simply replace the two lines below with:
-    return auth.wiki()
+    Controller for the home page. Returns SQLFORM of devices if logged in and a list of all the devices.
     """
+    sign_uuid = gluon_utils.web2py_uuid()
     if auth.user_id is None:
-        response.flash = T("Please sign in!")
-    return dict(message=T('Welcome to web2py!'))
+        # Would sign_uuid really be needed here? Leaving just in case.
+        return dict(message=T('Please sign in!'), sign_uuid=sign_uuid)
+    else:
+        device_list = db().select(db.devices.ALL)
+        return dict(device_list=device_list, sign_uuid=sign_uuid)
+
+
+@auth.requires_login()
+def add():
+    db.devices.device_id.writable = True
+    db.devices.last_sync.readable = db.devices.last_sync.writable = False
+    form = SQLFORM(db.devices)
+    if form.process().accepted:
+        session.flash = "Device added!"
+        redirect(URL('default', 'index'))
+    return dict(form=form)
+
+
+def load_devices():
+    # TODO: Condense "rows" to just for that one specific user instead of ALL devices
+    rows = db(db.devices).select()
```

Comment:
Yes, can you integrate here the work done  by the permissions team to display only devices the user has access to? 
This way we can also see if the permissions are well designed..